### PR TITLE
Added terraform files for S3 creation, iam user, outputs and variables.

### PR DESCRIPTION
**Pull request by jojuolape**
I have created terraform files 
S3 bucket creation and relevant Iam user accounts.
Variables to define values that user can input.
Outputs for relevant information regarding resources created. 

S3 bucket creation will allow user to define variables that will determine the properties of the bucket.
Variables defined will be bucket name, bucket access control, region bucket will reside and versioning.
The bucket resource terraform creates will be bucket_name prefixed with team_name. The bucket resource will have encryption and allow for force destroy as default. with no user input.

The user created will be named after the bucket resource name (team_name-bucket_name), this account will create an access key and secret key, which are also defined as outputs and i defined a policy to attach to this user account.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    ]
+
+resources = [
+    "arn:aws:s3:::${aws_s3_bucket.bucket.bucket}/*",
+    ]
+  }
+}
+
+resource "aws_iam_policy" "policy" {
+    name        = "${var.team_name}-s3-policy"
```
[the remark this answers is not in the document]
Comment (kalbir) [reply main.tf]: Sorry this was probably me being unclear -- I was thinking `team_name`-`bucket_name` i.e. use both of them.

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+         }
+     }
+    versioning {
+        enabled = "${var.versioning}"
+     }
+
+}
+
+resource "aws_iam_user" "s3-account" {
+    name = "${aws_s3_bucket.bucket.bucket}-s3-account"
```
[the remark this answers is not in the document]
Comment (jojuolape) [reply main.tf]: changed

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "arn:aws:s3:::${aws_s3_bucket.bucket.bucket}/*",
+    ]
+  }
+}
+
+resource "aws_iam_policy" "policy" {
+    name        = "${var.team_name}-s3-policy"
+    path        = "/teams/${var.team_name}/"
+    policy      = "${data.aws_iam_policy_document.policy.json}"
+    description = "S3 policy for team ${var.team_name}"
```
[the remark this answers is not in the document]
Comment (jojuolape) [reply main.tf]: changed

File: output.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    description  = "ARN for the new policy"
+    value        = "${aws_iam_policy.policy.arn}"   
+}
+
+output "iam_am_user" {
+    description  = "user name for s3 service account"
+    value        = "${aws_iam_user.s3-account.name}"
+}
+
+output "access_keys_id" {
```
[the remark this answers is not in the document]
Comment (jojuolape) [reply output.tf]: changed

File: output.tf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+output "bucket_arn" {
+    description   = "Arn for s3 bucket created"
+    value         = "${aws_s3_bucket.bucket.arn}"
+}
+
+output "bucket_name" {
+    description   = "bucket name"
+    value         = "${aws_s3_bucket.bucket.id}"
```
[the remark this answers is not in the document]
Comment (jojuolape) [reply output.tf]: changed

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    statement {
+        actions = [
+            "s3:ListBucket",
+            "s3:GetBucketLocation",
+            "s3:GetBucketVersioning",
+            "s3:GetObject",
+            "s3:CreateBucket",
+            "s3:DeleteBucket",
+            "s3:PutObject",
+    ]
```
[the remark this answers is not in the document]
Comment (jojuolape) [reply main.tf]: changed.

**Pull request merged**